src/app-components/table/Table.tsx

```diff
@@ -0,0 +1,161 @@
+  actionButtonHeader?: React.ReactNode;
+  /** Displays table in mobile mode */
+  mobile?: boolean;
+  size?: 'sm' | 'md' | 'lg';
+  zebra?: boolean;
+}
+
+function formatIfDate(value: unknown): string {
+  if (typeof value === 'string') {
+    const parsedDate = parseISO(value);
+    if (isValid(parsedDate)) {
+      return format(parsedDate, 'dd.MM.yyyy');
+    }
+  }
+  return String(value);
+}
+
+export function AppTable<T extends object>({
+  caption,
+  data,
+  columns,
+  actionButtons,
+  mobile,
+  actionButtonHeader,
+  size,
+  zebra,
+}: DataTableProps<T>) {
+  const defaultButtonVariant = mobile ? 'secondary' : 'tertiary';
+  return (
+    <Table
+      size={size || 'sm'}
+      className={cn(classes.table, { [classes.mobileTable]: mobile })}
+      zebra={zebra}
+    >
+      {caption}
+      <Table.Head>
+        <Table.Row>
+          {columns.map((col, index) => (
+            <Table.HeaderCell key={index}>{col.header}</Table.HeaderCell>
+          ))}
+
+          {actionButtons && actionButtons.length > 0 && (
+            <Table.HeaderCell>
+              <span className={classes.visuallyHidden}>{actionButtonHeader}</span>
+            </Table.HeaderCell>
+          )}
+        </Table.Row>
+      </Table.Head>
+      <Table.Body>
+        {data.map((rowData, rowIndex) => (
+          <Table.Row key={rowIndex}>
+            {columns.map((col, colIndex) => {
+              const cellValues = col.accessors
+                .filter((accessor) => !!pick(accessor, rowData))
+                .map((accessor) => pick(accessor, rowData));
+              if (cellValues.every((value) => value == null)) {
+                return (
+                  <Table.Cell
+                    key={colIndex}
+                    data-header-title={col.header}
+                  >
+                    -
+                  </Table.Cell>
+                );
+              }
+
+              if (col.renderCell) {
+                return (
+                  <Table.Cell
+                    key={colIndex}
+                    data-header-title={col.header}
+                  >
+                    {col.renderCell(cellValues, rowData)}
+                  </Table.Cell>
+                );
+              }
+
+              if (cellValues.length === 1) {
+                return (
+                  <Table.Cell
+                    data-header-title={col.header}
+                    key={colIndex}
+                  >
+                    {cellValues.map(formatIfDate)}
+                  </Table.Cell>
+                );
+              }
+
+              return (
+                <Table.Cell
+                  key={colIndex}
+                  data-header-title={col.header}
+                >
+                  <ul>
+                    {cellValues.map(formatIfDate).map((value, idx) => (
+                      <li key={idx}>{value}</li>
+                    ))}
+                  </ul>
+                </Table.Cell>
+              );
+            })}
```

Comment:
Might be a matter of taste, but I would prefer to extract large chunks like this into one or more separate components.